python/Dockerfile

```diff
@@ -55,76 +43,49 @@ USER dependabot
 ENV DEPENDABOT_NATIVE_HELPERS_PATH="/opt"
 RUN git -c advice.detachedHead=false clone https://github.com/pyenv/pyenv.git --branch $PYENV_VERSION --single-branch --depth=1 /usr/local/.pyenv
 
-FROM python-core as python-3.10
-RUN pyenv install $PY_3_10 \
-  && bash /opt/python/helpers/build $PY_3_10 \
+FROM python-core as python-3.8
+RUN pyenv install $PY_3_8 \
+  && bash /opt/python/helpers/build $PY_3_8 \
   && cd /usr/local/.pyenv \
-  && tar czf 3.10.tar.gz versions/$PY_3_10
+  && tar czf 3.8.tar.gz versions/$PY_3_8
 
 FROM python-core as python-3.9
 RUN pyenv install $PY_3_9 \
   && bash /opt/python/helpers/build $PY_3_9 \
   && cd /usr/local/.pyenv \
   && tar czf 3.9.tar.gz versions/$PY_3_9
 
-FROM python-core as python-3.8
-RUN pyenv install $PY_3_8 \
-  && bash /opt/python/helpers/build $PY_3_8 \
+FROM python-core as python-3.10
+RUN pyenv install $PY_3_10 \
+  && bash /opt/python/helpers/build $PY_3_10 \
   && cd /usr/local/.pyenv \
-  && tar czf 3.8.tar.gz versions/$PY_3_8
+  && tar czf 3.10.tar.gz versions/$PY_3_10
 
-FROM ghcr.io/dependabot/dependabot-updater-core
-ARG PY_3_11
-ARG PYENV_VERSION
+FROM python-core
+RUN pyenv install $PY_3_11 \
+  && pyenv global $PY_3_11 \
+  && bash /opt/python/helpers/build $PY_3_11
+
+COPY --from=python-3.10 /usr/local/.pyenv/3.10.tar.gz /usr/local/.pyenv/3.10.tar.gz
+COPY --from=python-3.9 /usr/local/.pyenv/3.9.tar.gz /usr/local/.pyenv/3.9.tar.gz
+COPY --from=python-3.8 /usr/local/.pyenv/3.8.tar.gz /usr/local/.pyenv/3.8.tar.gz
+
+# Install C-libs needed to build users' Python packages. Please document why each package is needed.
 USER root
 RUN apt-get update \
   && apt-get upgrade -y \
   && apt-get install -y --no-install-recommends \
-    dirmngr \
-    zlib1g-dev \
-    liblzma-dev \
-    libgdbm-dev \
-    bison \
-    tzdata \
-    zip \
-    openssh-client \
-    build-essential \
-    make \
-    libpq-dev \
-    libssl-dev \
-    libbz2-dev \
-    libffi-dev \
-    libreadline-dev \
-    libsqlite3-dev \
+    # Used by pycurl
     libcurl4-openssl-dev \
-    llvm \
-    libncurses5-dev \
-    libncursesw5-dev \
+    # Used by mysqlclient
     libmysqlclient-dev \
-    xz-utils \
-    tk-dev \
-    libxml2-dev \
-    libxmlsec1-dev \
-    libgeos-dev \
-    python3-enchant \
+    # Used by psycopg Postgres Client
+    libpq-dev \
+    # Used by python zoneinfo core lib
+    tzdata \
```

Comment:
This one is probably optional, `zoneinfo` cleanly falls back to the python package `tzinfo` if a system package doesn't provide it.